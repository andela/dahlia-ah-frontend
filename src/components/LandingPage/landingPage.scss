/*
* landingPage.scss 
* 
* Dahlia Frontend - Andela Simulations Challenge ©️ 2019
* Andela Simulations Challenge ©️ 2019
* https://github.com/andela/dahlia-ah-frontend
*
* SCSS Designed by TeamDahlia.
*/

$grey: #e5e5e5;
$white: #ffffff;
$black: #000008;

.landingPage {

  /* Writing Caption */
  #writingCaption {
    min-height:100%;
    background-size:cover;
    background:url(https://res.cloudinary.com/allebd/image/upload/v1567024807/dahlia/Rectangle_4.png) rgba(0,0,0,0.3);
    background-blend-mode: overlay;
    width: 100%;
    text-align: center;
    color: $white;
    padding: 45px;

    p {
      font-size: 42px;
      padding-left: 200px;
      padding-right: 200px;
    }
  }

  /* Membership Sections */
  #membership {
    display: grid;
    grid-template-rows: auto;
    grid-template-areas: "membership-title"
                            "membership-box";
    background: $grey;
    align-self: center;
    text-align: center;
    width: 100%;
    padding-top: 5px;
    padding-bottom: 5%;

    h2 {
      grid-area: membership-title;
      font-weight: bold;
      text-transform: uppercase;
      font-size: 36px;
    }

    h4 {
      margin: 0;
      text-transform: uppercase;
      font-size: 24px;
      padding: 10px 0;
      border-bottom: 1px solid $white;
    }

    .membershipButton {
      height: 50px;
      cursor: pointer;

      p {
        margin: 0;
        cursor: pointer;
        padding-top: 15px;
      }
    }

    .subscribeMembership h4 {
      border-bottom: 1px solid $black;
    }
  }

  #membership ul, #membership li {
    padding: 5px;
  }

  .membershipBox {
    grid-area: membership-box;
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    width: 550px;
    justify-self: center;
  }

  .freeMembership {
    border: 1px solid $black;
    background: $white;
  }

  .freeMembership div:first-child {
    height: 300px;
    background: $black;
    color: $white;
  }

  .freeMembership div:first-child ul {
    margin-top: 65px;
  }

  .freeMembership button, .subscribeMembership button {
    border: 0;
    background: #00000000;
    cursor: pointer;
  }

  .subscribeMembership {
    background: #eeeeee;
    box-shadow: 0px 4px 46px rgba(0, 0, 0, 0.25);
  }

  .subscribeMembership div:first-child {
    height: 300px;
    background: $white;
  }

  .subscribeMembership div:first-child ul {
    margin-top: 30px;
  }

  /* buttons */
  button.btnLarge {
    color: $white;
    border: 2px solid $white;
    padding: 10px 20px;
    font-size: 20px;
    font-weight: 400;
    text-transform: uppercase;
    background-color: rgba(0,0,0,0);
    cursor: pointer;
  }

  button.btnMedium {
    color: $black;
    border: 2px solid $black;
    padding: 10px 20px;
    font-size: 16px;
    font-weight: 400;
    text-transform: uppercase;
    background-color: rgba(0,0,0,0);
    cursor: pointer;
  }

  @media(max-width: 900px){
    #writingCaption p {
      padding-left: 0;
      padding-right: 0;
    }

    .membershipBox {
      width: 400px;
    }
  }

  @media(max-width: 500px){
    .membershipBox {
      grid-template-columns: 1fr;
      grid-gap: 40px;
      width: 300px;
    }
  }
}
