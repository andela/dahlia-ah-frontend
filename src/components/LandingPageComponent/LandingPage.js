import React from 'react';
import { Link } from 'react-router-dom';
import WeeklyNovelComponent from '../WeeklyNovelComponent/WeeklyNovel';
import MembershipComponent from '../MembershipComponent/Membership';
import style from './landingPage.scss';

const LandingPage = () => (
  <div>
    <section id={style.banner}>
      <div className={style.bannerGradient}>
        <h2>THE MAN IN THE FOREST OF DREAMS</h2>
        <div className={style.bannerText}>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,
            sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco
            laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure
            dolor in reprehenderit in voluptate velit esse cillum dolore
            eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
            non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
        <button type="button" className={style.btnLarge}><Link to="/">View Snippet</Link></button>
      </div>
    </section>
    <WeeklyNovelComponent />
    <section id={style.writingCaption}>
      <p>The Best Place To Express Yourself and Collaborate</p>
      <button type="button" className={style.btnLarge}><Link to="/">Start Writing</Link></button>
    </section>
    <MembershipComponent />
  </div>
);

export default LandingPage;
