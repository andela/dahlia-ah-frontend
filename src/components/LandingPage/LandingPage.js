import React from 'react';
import { Link } from 'react-router-dom';
import './landingPage.scss';

const LandingPage = () => (
  <div>
    <section id="banner">
      <div className="bannerGradient">
        <h2>THE MAN IN THE FOREST OF DREAMS</h2>
        <div className="bannerText">
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit,
            sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco
            laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure
            dolor in reprehenderit in voluptate velit esse cillum dolore
            eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
            non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </div>
        <button type="button" className="btnLarge"><Link to="/">View Snippet</Link></button>
      </div>
    </section>
    <section id="weekNovel">
      <h2>Novel of the Week</h2>
      <div className="weekNovelBox">
        <div className="weekNovelImage">
          <img src="https://res.cloudinary.com/allebd/image/upload/v1567031973/dahlia/weeklynovel.png" alt="weekly novel" />
        </div>
        <div className="weekNovelDescription">
          <div>
            <h3>One Way Trip</h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit,
              sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
              Ut enim ad minim veniam, quis nostrud exercitation ullamco
              laboris nisi ut aliquip ex ea commodo consequat.
              Duis aute irure dolor in reprehenderit in voluptate velit esse
              cillum dolore eu fugiat nulla pariatur.
              Excepteur sint occaecat cupidatat non proident,
              sunt in culpa qui officia deserunt mollit anim id est laborum..
              Duis aute irure dolor in reprehenderit in voluptate velit esse
              cillum dolore eu fugiat nulla pariatur.
              Excepteur sint occaecat cupidatat non proident,
              sunt in culpa qui officia deserunt mollit anim id est laborum.
              Excepteur sint occaecat cupidatat non proident,
              sunt in culpa qui officia deserunt mollit anim id est
              laborum upidatat non proident, sunt in culpa qui officia.
            </p>
            <button type="button" className="btnMedium"><Link to="/">View Snippet</Link></button>
          </div>
        </div>
      </div>
    </section>
    <section id="writingCaption">
      <p>The Best Place To Express Yourself and Collaborate</p>
      <button type="button" className="btnLarge"><Link to="/">Start Writing</Link></button>
    </section>
    <section id="membership">
      <h2>Membership</h2>
      <div className="membershipBox">
        <div className="freeMembership">
          <div>
            <h4>Free</h4>
            <ul>
              <li>Read Novels</li>
              <li>Bookmark Novels</li>
              <li>Follow Authors</li>
            </ul>
          </div>
          <div>
            <p><Link to="/">SIGN UP</Link></p>
          </div>
        </div>
        <div className="subscribeMembership">
          <div>
            <h4>Subscription</h4>
            <ul>
              <li>Read Novels</li>
              <li>Bookmark Novels</li>
              <li>Follow and Collaborate with other Authors</li>
              <li>Like and Comment on Novels</li>
              <li>Novel Review and Analytics</li>
            </ul>
          </div>
          <div>
            <p><Link to="/">SUBSCRIBE</Link></p>
          </div>
        </div>
      </div>
    </section>
  </div>
);

export default LandingPage;
