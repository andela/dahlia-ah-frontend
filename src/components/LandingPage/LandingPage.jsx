import React from 'react';
import Banner from '../Banner/Banner';
import NovelOfTheWeek from '../NovelOfTheWeek/NovelOfTheWeek';
import './landingPage.scss';

const LandingPage = () => (
  <div>
    <div className="landingPage">
      <Banner />
      <NovelOfTheWeek />
      <section id="writingCaption">
        <p>The Best Place To Express Yourself and Collaborate</p>
        <button type="button" className="btnLarge">Start Writing</button>
      </section>
      <section id="membership">
        <h2>Membership</h2>
        <div className="membershipBox">
          <div className="freeMembership">
            <div>
              <h4>Free</h4>
              <ul>
                <li>Read Novels</li>
                <li>Bookmark Novels</li>
                <li>Follow Authors</li>
              </ul>
            </div>
            <div>
              <p>SIGN UP</p>
            </div>
          </div>
          <div className="subscribeMembership">
            <div>
              <h4>Subscription</h4>
              <ul>
                <li>Read Novels</li>
                <li>Bookmark Novels</li>
                <li>Follow and Collaborate with other Authors</li>
                <li>Like and Comment on Novels</li>
                <li>Novel Review and Analytics</li>
              </ul>
            </div>
            <div>
              <p>SUBSCRIBE</p>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
);

export default LandingPage;
