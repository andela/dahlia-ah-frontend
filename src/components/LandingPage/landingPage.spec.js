import React from 'react';
import { BrowserRouter, Switch } from 'react-router-dom';
import { configure, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import LandingPage from './LandingPage';

configure({ adapter: new Adapter() });

describe('App component', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <BrowserRouter>
        <Switch>
          <LandingPage />
        </Switch>
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
