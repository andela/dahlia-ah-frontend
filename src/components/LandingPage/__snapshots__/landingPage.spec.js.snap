// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage component renders correctly 1`] = `ReactWrapper {}`;
